source/s3_client.c

```diff
@@ -337,6 +337,7 @@ struct aws_s3_client *aws_s3_client_new(
         } else {
 #ifdef BYO_CRYPTO
             AWS_ASSERT(false);
+            goto on_error;
```

Comment:
Not sure we want to handle this error.
- If we want to crash in this case, use `AWS_FATAL_ASSERT(false);`
- If we want to handle it more properly, add `aws_raise_error(); /* Some proper error */`